File: 942-DI String Match-增减字符串匹配/942-DI String Match.java
class Solution {
    public int[] diStringMatch(String S) {
        char s[] = S.toCharArray();
        int minNum = 0;
        int maxNum = S.length();
        int[] newArray = new int[S.length()+1];
        int index = 0 ;
        for(int i = 0 ;i < S.length() ;i++){
          if(s[i] == 'D'){
            newArray[i] = maxNum;
            maxNum--;
          }else{
            newArray[i] = minNum;
            minNum++;
          }
          newArray[i+1] = maxNum;
        }
        return newArray;
    }
}

File: 942-DI String Match-增减字符串匹配/942-DI String Match.js
/**
 * @param {string} S
 * @return {number[]}
 */
var diStringMatch = function(S) {
    let maxNum = S.length;
    let minNum = 0;
    let newArray = [];
    for(let i = 0 ;i < S.length ; i++){
      if(S[i] === 'I'){
        newArray.push(minNum);
        minNum++;
      }else if(S[i] === 'D'){
        newArray.push(maxNum);
        maxNum--;
      }
    }
      newArray.push(maxNum)
    return newArray;
};
